src/helpers/auth.js

```diff
@@ -82,7 +82,8 @@ class PegaAuth {
           this.buildAuthorizeUrl(state).then((url) => {
               let myWindow = null; // popup or iframe
               let elIframe = null;
-              let bWinIframe = (!!this.config.userIdentifier && !!this.config.password) || this.config.authService !== "pega";
+              const iframeTimeout = this.config.silentTimeout !== undefined ? this.config.silentTimeout : 5000;
+              let bWinIframe = iframeTimeout > 0 && ((!!this.config.userIdentifier && !!this.config.password) || this.config.authService !== "pega");
```

Comment:
Just curious about the use of !!
That's a new usage to me. What's the intent of that?

Reply:
Found this post which articulates how it is a way with minimal characters to do a boolean conversion and handle various null/undefined/empty string check scenarios.  https://www.sitepoint.com/javascript-double-negation-trick-trouble/

So in these cases using it for the property exists and is not empty